### PR TITLE
Run nym-vpnd in background service on windows

### PR DESCRIPTION
**Pull request by octol**
# Overview

Add support for running nym-vpnd as a windows service. The service is registered with the windows event log and log messages are sent there.

# How to use

1. First register the service
```
./nym-vpnd --install
```
2. Then start the service in the Windows Service control app

# Future

This is the first PR towards fully supporting running as a background service. 
Known issues:
- Stop is reported as failed in the UI, but will succeed after awhile if you refresh the service status
- The handler controlling the vpn lib is not yet started as it fails the service to start
- The gPRC listener is functional, but cannot interact with the vpn lib yet, and as such all grpc messages will timeout

### CODE REVIEW

File: nym-vpn-core/nym-vpnd/src/win_service/service.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+windows_service::define_windows_service!(ffi_service_main, my_service_main);
+
+pub(crate) static SERVICE_NAME: &str = "nym-vpnd";
+pub(crate) static SERVICE_DISPLAY_NAME: &str = "NymVPN Service";
+
+pub(crate) static SERVICE_DESCRIPTION: &str =
+    "A service that creates and runs tunnels to the Nym network";
+static SERVICE_TYPE: ServiceType = ServiceType::OWN_PROCESS;
+
+fn my_service_main(arguments: Vec<OsString>) {
```
Comment (neacsu) [inline nym-vpn-core/nym-vpnd/src/win_service/service.rs:30]: This looks like an example type function name. Maybe something like `nym_service_main` instead?

File: nym-vpn-core/nym-vpnd/src/win_service/service.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+#[allow(unused)]
+#[derive(thiserror::Error, Debug)]
+pub enum InstallError {
+    #[error("Unable to connect to service manager")]
+    ConnectServiceManager(#[source] windows_service::Error),
+
+    #[error("Unable to create a service")]
+    CreateService(#[source] windows_service::Error),
+}
```
Comment (neacsu) [inline nym-vpn-core/nym-vpnd/src/win_service/service.rs:115]: Let's use a separate `error.rs` file for this, like we do in most of our codebase.

**Review by neacsu — APPROVED** [review]

File: nym-vpn-core/nym-vpnd/src/win_service/install.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    service_manager::{ServiceManager, ServiceManagerAccess},
+};
+use windows_sys::Win32::Foundation::ERROR_SERVICE_DOES_NOT_EXIST;
+
+// see https://github.com/mullvad/windows-service-rs/blob/main/examples/install_service.rs
+pub(super) fn install_service() -> windows_service::Result<()> {
+    let manager_access = ServiceManagerAccess::CONNECT | ServiceManagerAccess::CREATE_SERVICE;
+    let service_manager = ServiceManager::local_computer(None::<&str>, manager_access)?;
+
+    println!("Registering event logger {}...", SERVICE_DISPLAY_NAME);
```
[the remark this answers is not in the document]
Comment (octol) [reply nym-vpn-core/nym-vpnd/src/win_service/install.rs:17]: Yep agree

File: nym-vpn-core/nym-vpnd/src/win_service/service.rs (hunk 1 above)
Comment (octol) [reply nym-vpn-core/nym-vpnd/src/win_service/service.rs:30]: Yeah this is leftover from example code, I'll change it!

**Pull request merged**